app/src/main/java/com/uniandes/vynilos/presentation/ui/screen/ArtistScreen.kt

```diff
@@ -0,0 +1,178 @@
+    //navigationActions: NavigationActions = NavigationActions()
+) {
+    val artistsResult by viewModel.artistResult.collectAsState()
+
+    var errorMessage by remember { mutableStateOf("") }
+
+    LaunchedEffect(Unit) {
+        viewModel.getArtists()
+    }
+
+
+    viewModel.artistResult.observeAsActions {
+        if (it is DataState.Error) {
+            errorMessage = it.error.message
+        }
+    }
+
+
+    VynilOSTheme {
+        Scaffold(modifier = Modifier.fillMaxSize()) { _ ->
+            Box(
+                modifier = Modifier
+                    .fillMaxSize(),
+                contentAlignment = Alignment.TopCenter
+            ) {
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    modifier = Modifier.fillMaxSize()
+                ) {
+                    Text(
+                        text = stringResource(R.string.artists),
+                        modifier = Modifier.padding(16.dp)
+                    )
+
+                    Spacer(modifier = Modifier.height(16.dp))
+
+                    when (val result = artistsResult) {
+                        is DataState.Loading -> {
+                            CircularProgressIndicator(
+                                modifier = Modifier.size(100.dp).align(Alignment.CenterHorizontally)
+                            )
+                        }
+                        is DataState.Success -> {
+                            val data = result.getDataOrNull()
+                            if (!data.isNullOrEmpty()) {
+                                LazyColumn(
+                                    modifier = Modifier.fillMaxSize().padding(16.dp),
+                                    verticalArrangement = Arrangement.spacedBy(8.dp),
+                                    horizontalAlignment = Alignment.CenterHorizontally
+                                ) {
+                                    items(data) { artist ->
+                                        ArtistCard(artist)
+                                    }
+                                }
+                            } else {
+                                Text(
+                                    text = stringResource(R.string.no_artist),
+                                    color = Color.Gray,
+                                    modifier = Modifier.align(Alignment.CenterHorizontally)
+                                )
+                            }
+                        }
+                        is DataState.Error -> {
+                            Text(
+                                text = errorMessage,
+                                color = Color.Red,
+                                modifier = Modifier.align(Alignment.CenterHorizontally)
+                            )
+                        }
+                        else -> {
+                            // En caso de que no haya ningún estado particular
```

Comment:
Si encierra todo el when en un `apply` no tendrías que agregar el else 

`
artistResult?.apply {
                    when(val result = this) {
                        is DataState.Loading -> {
  
                        }
                        is DataState.Success -> {
                            
                        }
                        is DataState.Error -> {
           
                        }
                    }
                }


`